File: backend/classify.py
# Mirrors src/lib/classify.ts — same signals, same priority (spam beats toxic).
# 'code' and 'click' removed: too broad, caused false positives on
# "game code is buggy" and "click to attack".

SPAM_SIGNALS = [
    "discount", "promo code", "coupon code", "click here",
    "free v-bucks", "subscribe", "link in bio", "giveaway",
]

TOXIC_SIGNALS = [
    "stupid", "trash", "garbage", "idiot",
    "worst game ever", "devs are idiots",
]


def classify_by_rules(text: str) -> dict:
    lower = text.lower()
    if any(s in lower for s in SPAM_SIGNALS):
        return {"classification": "spam", "reason": "Detected promotional/spam language"}
    if any(s in lower for s in TOXIC_SIGNALS):
        return {"classification": "toxic", "reason": "Detected toxic language"}
    return {"classification": "helpful", "reason": "Rule-based classification (no AI key configured)"}

File: src/lib/classify.ts
export type ClassificationResult = {
  classification: 'helpful' | 'spam' | 'toxic';
  reason: string;
};

// 'code' and 'click' removed — too broad, caused false positives on
// phrases like "cheat codes", "error code", "click to attack".
const SPAM_SIGNALS = [
  'discount', 'promo code', 'coupon code', 'click here',
  'free v-bucks', 'subscribe', 'link in bio', 'giveaway',
];

const TOXIC_SIGNALS = [
  'stupid', 'trash', 'garbage', 'idiot',
  'worst game ever', 'devs are idiots',
];

export function classifyByRules(text: string): ClassificationResult {
  const lower = text.toLowerCase();

  if (SPAM_SIGNALS.some((s) => lower.includes(s))) {
    return { classification: 'spam', reason: 'Detected promotional/spam language' };
  }
  if (TOXIC_SIGNALS.some((s) => lower.includes(s))) {
    return { classification: 'toxic', reason: 'Detected toxic language' };
  }
  return { classification: 'helpful', reason: 'Rule-based classification (no AI key configured)' };
}
